lib/astutils.cpp

```diff
@@ -2407,7 +2407,7 @@ bool isVariableChangedByFunctionCall(const Token *tok, int indirect, const Setti
             continue;
         conclusive = true;
         if (indirect > 0) {
-            if (!arg->isConst() && arg->isPointer())
+            if (arg->isPointer() && !(arg->valueType() && arg->valueType()->pointer == indirect && (arg->valueType()->constness & (1 << (indirect - 1))) != 0))
```

Comment:
I think it would be `indirect - arg->valueType()->pointer`. I need to correct this in some other code I wrote.

Reply:
Can we add some function/macro for this?

Reply:
So I have tried `if (arg->isPointer() && !(arg->valueType() && (arg->valueType()->constness & (1 << (arg->valueType()->pointer - indirect))) != 0)) `
This causes the new test at testastutils.cpp:290 to fail:
~~~
"void g(int* const* pp);\n"
"void f(int* p) {\n"
"    return g(&p);\n"
"}\n"
~~~

Reply:
>  I think it would be `indirect - arg->valueType()->pointer`

Actually its backwards it should be `arg->valueType()->pointer - indirect`, since pointer is always greater or equal to `indirect`.

> Let me try to add a function in #4912.

I added an `isConst` method to `ValueType` which will do this calculation:

https://github.com/danmar/cppcheck/pull/4912/files#diff-304f15dd3a4d25ecfae37990291a91ca66aecd56342bcf2979c102afa4da262dR7417

So you can just write `arg->valueType()->isConst(indirect)`.